nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java

```diff
@@ -121,18 +147,47 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             if (createContainer && !containerClient.exists()) {
                 containerClient.create();
             }
+
             BlobClient blobClient = containerClient.getBlobClient(blobName);
+            final BlobRequestConditions blobRequestConditions = new BlobRequestConditions();
+            Map<String, String> attributes = new HashMap<>();
+            applyCommonAttributes(attributes, blobClient);
+
+            try {
+                if (conflictResolution != AzureStorageConflictResolutionStrategy.REPLACE_RESOLUTION) {
+                    blobRequestConditions.setIfNoneMatch("*");
+                }
+
+                try (InputStream rawIn = session.read(flowFile)) {
+                    final BlobParallelUploadOptions blobParallelUploadOptions = new BlobParallelUploadOptions(toFluxByteBuffer(rawIn));
+                    blobParallelUploadOptions.setRequestConditions(blobRequestConditions);
+                    Response<BlockBlobItem> response = blobClient.uploadWithResponse(blobParallelUploadOptions, null, Context.NONE);
+                    BlockBlobItem blob = response.getValue();
+                    long length = flowFile.getSize();
+                    applyUploadResultAttributes(attributes, blob, BlobType.BLOCK_BLOB, length);
 
-            long length = flowFile.getSize();
+                }
+            } catch (BlobStorageException e) {
+                final BlobErrorCode errorCode = e.getErrorCode();
+                session.putAttribute(flowFile, ATTR_NAME_ERROR_CODE, e.getErrorCode().toString());
 
-            try (InputStream rawIn = session.read(flowFile);
-                 BufferedInputStream bufferedIn = new BufferedInputStream(rawIn)) {
-                blobClient.upload(bufferedIn, length);
+                final boolean alreadyExists = errorCode == BlobErrorCode.BLOB_ALREADY_EXISTS;
+                final boolean ignore = conflictResolution == AzureStorageConflictResolutionStrategy.IGNORE_RESOLUTION;
+
+                // If the blob already exists, we always add an attribute "ignored"; depending on the value of
+                // conflict resolution, this will be true or false.
+                if (alreadyExists) {
+                    session.putAttribute(flowFile, ATTR_NAME_IGNORED, String.valueOf(ignore));
```

Comment:
The comment is not valid. When the resolution is `replace`, this attribute is not applied, though the blob can already exist. I'd only apply this attribute in case of `ignore` resolution and set the value depending on the presence of the file.

Reply:
Good point. I have changed this in 4a822e15df25ad5eb24ddf02d54a55799213a003 now.